use-cases/xtclim/src/anomaly.py

```diff
@@ -1,114 +1,182 @@
-                                                        pixel_wise_criterion)
-                tot_proj_losses = list(map(add, tot_proj_losses, proj_losses))
-                proj_avg_losses += proj_avg_loss
-            
-            tot_proj_losses = np.array(tot_proj_losses)/n_avg
-            proj_avg_losses = proj_avg_losses/n_avg
-
-            # save the losses time series
-            pd.DataFrame(tot_proj_losses).to_csv(f"../outputs/proj{scenario}_losses_{season}1d_allssp.csv")
-            print(f'SSP{scenario} Projection average loss:', proj_avg_losses, 'for', season[:-1])
\ No newline at end of file
+                train_avg_loss, _, train_losses, _ = evaluate(cvae_model, trainloader,
+                                                 trainset, self.device, criterion,
+                                                 pixel_wise_criterion)
+                tot_train_losses = list(map(add, tot_train_losses, train_losses))
+                train_avg_losses += train_avg_loss
+                test_avg_loss, _, test_losses, _ = evaluate(cvae_model, testloader,
+                                                 testset, self.device, criterion,
+                                                 pixel_wise_criterion)
+                tot_test_losses = list(map(add, tot_test_losses, test_losses))
+                test_avg_losses += test_avg_loss
+            tot_train_losses = np.array(tot_train_losses)/n_avg
+            tot_test_losses = np.array(tot_test_losses)/n_avg
+            train_avg_losses = train_avg_losses/n_avg
+            test_avg_losses = test_avg_losses/n_avg
+
+            pd.DataFrame(tot_train_losses).to_csv(f"outputs/train_losses_{season}1d_allssp.csv")
+            pd.DataFrame(tot_test_losses).to_csv(f"outputs/test_losses_{season}1d_allssp.csv")
+            print('Train average loss:', train_avg_losses)
+            print('Test average loss:', test_avg_losses)
+
+            self.log(train_avg_losses,
+                    'Average train loss',
+                    kind='metric')
+            self.log(test_avg_losses,
+                     'Average test loss',
+                     kind='metric')
+
+            # Clean-up
+            if self.strategy.is_main_worker and self.logger:
+                self.logger.destroy_logger_context()
+
+            self.strategy.clean_up()
+
+        else:
+            season = self.seasons
+            # load previously trained model
+            cvae_model = NNmodel.ConvVAE().to(self.device)
+            cvae_model.load_state_dict(self.model['model_state_dict'], strict=False)
+
+            distribute_kwargs = {}
+            # Distributed model, optimizer, and scheduler
+            cvae_model, _, _ = self.strategy.distributed(
+                cvae_model, **distribute_kwargs
+            )
+
+            if self.strategy.is_main_worker and self.logger:
+                 self.logger.create_logger_context()
+
+            for scenario in ['585', '245']:
```

Comment:
These are magic "numbers". Make them into constants or at least some variable that tells us what they mean.